rdbms/src/main/java/io/spine/server/storage/jdbc/TypeMappings.java

```diff
@@ -0,0 +1,100 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.jdbc;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.spine.server.storage.jdbc.TypeMapping.Builder;
+
+import java.sql.SQLException;
+
+import static io.spine.server.storage.jdbc.Type.BOOLEAN;
+import static io.spine.server.storage.jdbc.Type.BYTE_ARRAY;
+import static io.spine.server.storage.jdbc.Type.INT;
+import static io.spine.server.storage.jdbc.Type.LONG;
+import static io.spine.server.storage.jdbc.Type.STRING;
+import static io.spine.server.storage.jdbc.Type.STRING_255;
+
+/**
+ * Default {@link TypeMapping type mappings} for different databases.
+ *
+ * @author Dmytro Grankin
+ */
+public class TypeMappings {
+
+    private static final TypeMapping MY_SQL = baseMapping().add(BYTE_ARRAY, "BLOB")
+                                                           .build();
+    private static final TypeMapping POSTGRES = baseMapping().add(BYTE_ARRAY, "BYTEA")
+                                                             .build();
+
+    private TypeMappings() {
+        // Prevent instantiation of this utility class.
+    }
+
+    /**
+     * Obtains the default type mapping for MySQL database.
+     */
+    @VisibleForTesting
+    public static TypeMapping mySql() {
+        return MY_SQL;
+    }
+
+    /**
+     * Obtains the default type mapping for PostreSQL database.
+     */
+    @VisibleForTesting
+    public static TypeMapping postgreSql() {
+        return POSTGRES;
+    }
+
+    /**
+     * Obtains the type mapping for the used
+     * {@linkplain java.sql.DatabaseMetaData#getDatabaseProductName() database}.
+     * 
+     * @param dataSource the data source to get database metadata
+     * @return the type mapping for the used database
+     *         or {@linkplain #mySql() MySQL} mapping if there is no standard mapping for the database
+     */
+    static TypeMapping get(DataSourceWrapper dataSource) {
+        try (final ConnectionWrapper connection = dataSource.getConnection(true)) {
+            final String name = connection.get()
+                                          .getMetaData()
+                                          .getDatabaseProductName()
+                                          .toLowerCase();
+            if ("postgresql".equals(name)) {
+                return POSTGRES;
+            }
+
+            return MY_SQL;
+        } catch (SQLException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    private static TypeMapping.Builder baseMapping() {
```

Comment:
As long as this method is called at least twice, it has to have a smarter implementation. And not do the same things over and over again.